systemd/systemd.go

```diff
@@ -1457,23 +1484,24 @@ func (s *systemd) AddMountUnitFileWithOptions(unitOptions *MountUnitOptions) (st
 	daemonReloadLock.Lock()
 	defer daemonReloadLock.Unlock()
 
-	mountUnitName, err := writeMountUnitFile(unitOptions)
+	mountUnitName, modified, err := writeMountUnitFile(unitOptions)
 	if err != nil {
 		return "", err
 	}
+	if modified != MountClean {
```

Comment:
I would expect this change to have a major effect on our unit tests, but it looks like the only reason they had to be changed was the switch from `Start` to `ReloadOrRestart`, which makes me wonder if the test coverage here is good enough. But in any case, it would be nice to have a unit test that exercises the code path when `modified` is equal to `MountClean`.

Reply:
In the case the unit was created, the service will not be started and in this case `ReloadOrRestart` will have the same behavior as `Start`. So this is why I did not split the case `MountUpdated` and `MountCreated`.

That the case `MountClean` is not tested in unit test was a clear oversight however. I will add it.

Reply:
The test is `TestEnureMountUnitUnchanged`.